lib/ansible/modules/cloud/vmware/vmware_vspan_session.py

```diff
@@ -0,0 +1,582 @@
+        self.switch = module.params['switch']
+        self.name = module.params['name']
+        self.session_type = module.params['session_type']
+        self.enabled = module.params['enabled']
+        self.state = module.params['state']
+        self.description = module.params['description']
+        self.source_port_transmitted = module.params['source_port_transmitted']
+        self.source_port_received = module.params['source_port_received']
+        self.destination_port = module.params['destination_port']
+        self.encapsulation_vlan_id = module.params['encapsulation_vlan_id']
+        self.strip_original_vlan = module.params['strip_original_vlan']
+        self.mirrored_packet_length = module.params['mirrored_packet_length']
+        self.normal_traffic_allowed = module.params['normal_traffic_allowed']
+        self.sampling_rate = module.params['sampling_rate']
+        self.dv_switch = find_dvs_by_name(self.content, self.switch)
+        self.operation = None
+        self.modified_ports = dict()
+        self.deleted_session = None
+        if module.params['source_vm_transmitted'] is not None:
+            self.source_vm_transmitted_name = module.params['source_vm_transmitted']['name']
+            self.source_vm_transmitted_nic_label = module.params['source_vm_transmitted']['nic_label']
+        if module.params['source_vm_received'] is not None:
+            self.source_vm_received_name = module.params['source_vm_received']['name']
+            self.source_vm_received_nic_label = module.params['source_vm_received']['nic_label']
+        if module.params['destination_vm'] is not None:
+            self.destination_vm_name = module.params['destination_vm']['name']
+            self.destination_vm_nic_label = module.params['destination_vm']['nic_label']
+
+    def set_operation(self):
+        """Sets the operation according to state"""
+        if self.state == 'absent':
+            self.operation = 'remove'
+        elif self.state == 'present' and self.find_session_by_name() is None:
+            self.operation = 'add'
+        else:
+            self.operation = 'edit'
+
+    def find_session_by_name(self):
+        """Finds a session by name
+
+        Returns
+        -------
+        vim.dvs.VmwareDistributedVirtualSwitch.VspanSession
+            The session if there was a session by the given name, else returns None
+        """
+        for vspan_session in self.dv_switch.config.vspanSession:
+            if vspan_session.name == self.name:
+                return vspan_session
+        return None
+
+    def get_vm_port(self, vm_name, nic_label):
+        """Finds the port of the VM
+
+        Returns
+        -------
+        str
+            the port number as a string, or None if the NIC couldnt be found
+        """
+        vm = find_vm_by_name(self.content, vm_name)
+        if vm is None:
+            self.module.fail_json(msg='There is no VM with this name.')
```

Comment:
I think it would be good to mention the name, so people know which name is failing...
```suggestion
            self.module.fail_json(msg="There is no VM with the name: {0:s}.".format(vm_name))
```